crates/services/src/async_processor.rs

```diff
@@ -269,29 +272,34 @@ mod tests {
 
         // Then
         while broadcast_receiver.recv().await.is_ok() {}
-        assert!(instant.elapsed() >= Duration::from_secs(10));
+        // 5 tasks running on 1 thread, each task taking 1 second,
+        // should complete in approximately 5 seconds overall.
+        // Allowing some LEEWAY to account for runtime overhead.
+        const LEEWAY: Duration = Duration::from_millis(300);
+        assert!(instant.elapsed() < Duration::from_secs(5) + LEEWAY);
```

Comment:
If the test complete them all in 1 seconds because it spawned too many thread it's an issue I think an other assert could be added to ensure it : 
```suggestion
        assert!(instant.elapsed() < Duration::from_secs(5) + LEEWAY);
        assert!(instant.elapsed() > Duration::from_secs(5));
```
I don't think thr `LEEWAY` is necessary here because it should never take less than 5 seconds in my mind but if you think there is an expected case then i'm ok to add the ` - LEEWAY` in second assert.

Reply:
I agree with the additional assert, added in this commit: 8443c00eccaef13b6e607685ceb102b0d8a4fad4
However, I used `>=` to cover the super unlikely case that the execution took exactly 5 secs.

Reply:
> I don't think thr LEEWAY is necessary here because it should never take less than 5 seconds in my mind but if you think there is an expected case then i'm ok to add the  - LEEWAY in second assert.

Yes, it will never take less that 5 secs. on traditional computers :-)
So the `elapsed` could be, let's say 5.11 sec, and I wanted to check if it's not bigger, let's say, 7 sec, because ultimately the correct behavior of this test is to execute for just slightly more than 5 secs.

I could do
```
assert!(instant.elapsed() >= Duration::from_secs(5));
assert!(instant.elapsed() < Duration::from_secs(6));
```
but I don't like the additional magic number (6) and I thought that the concept of leeway would be more clear.

Let me know if this makes any sense in the context of async processor :)

Reply:
yeah yeah i like the leeway in the first assert :)